src/vs/workbench/api/node/extHostTypes.ts

```diff
@@ -491,10 +491,29 @@ export class TextEdit {
 	}
 }
 
-export class WorkspaceEdit {
+export class WorkspaceEdit implements vscode.WorkspaceEdit {
 
-	private _values: [URI, TextEdit[]][] = [];
-	private _index = new Map<string, number>();
+	private _clock: number = 0;
+
+	private _resourceEdits: [number/*time*/, URI, URI][] = [];
```

Comment:
Same comment about avoid tuples in our internals as well. They are weird type system wise (`typeof([1, 2]) !== type [number, number]`) and it easy to confuse tuple fields when writing or refactoring code

Reply:
@mjbvz agree.

Reply:
Yeah, can make that change internally.. For the API I'd go for consistency even tho I share the concern. Having said that, I don't expect heavy usage of those reading APIs